main.go

```diff
@@ -19,47 +19,60 @@ import (
 	"github.com/urfave/cli/v2/altsrc"
 )
 
-const redColor = "\u001B[31m"
-const greenColor = "\u001B[32m"
-const blueColor = "\u001B[36m"
-const endColor = "\u001B[0m"
-
-// Colorful prefixes
-const ttyConnectedPrefix = greenColor + "●" + endColor
-const ttyDisconnectedPrefix = redColor + "●" + endColor
-const ttyInfoPrefix = blueColor + "●" + endColor
-const ttyErrorPrefix = redColor + "!" + endColor
-
-// Black & white prefixes. Unicode characters
-const bwConnectedPrefix = "✓"
-const bwDisconnectedPrefix = "𐄂"
-const bwErrorPrefix = "!"
-const bwInfoPrefix = "·"
-
 type LogMessage struct {
 	level   log.Level
 	message error
 }
 
-// UserInterfacePreferences helps to hold UI preferences
-type UserInterfacePreferences struct {
-	IsColorful         bool
-	MachineReadable    bool
-	ConnectedPrefix    string
-	DisconnectedPrefix string
-	ErrorPrefix        string
-	InfoPrefix         string
+// userInterfaceSettings manages standard output preference.
+// It tracks colors, icons and machine-readable output (e.g. json).
+//
+// It is instantiated via uiSettings by calling configureUISettings.
+type userInterfaceSettings struct {
+	// isMachineReadable describes the machine-readable mode (e.g., `--format json`)
+	isMachineReadable bool
+	// isRich describes the ability to display colors and animations
+	isRich    bool
+	iconOK    string
+	iconInfo  string
+	iconError string
+}
+
+// uiSettings is an instance that keeps actual data of output preference.
+//
+// It is managed by calling the configureUISettings method.
+var uiSettings = userInterfaceSettings{}
+
+// configureUISettings is called by the CLI library when it loads up.
+// It sets up the uiSettings object.
+func configureUISettings(ctx *cli.Context) {
+	if ctx.Bool("no-color") {
+		uiSettings = userInterfaceSettings{
+			isRich:            false,
+			isMachineReadable: false,
+			iconOK:            "✓",
```

Comment:
We also received UX feedback saying that these cool icons are not so cool for blind people using screen readers. Maybe we should replace it with something readable by screen readers. I'm not telling that it should be part of this PR, but the name `iconOK` will likely have to change in the future.

Reply:
ACK. I have just checked with Orca:
- the colored options are all read as "circle"
- the `--no-color` options are skipped (𐄂, !) or read aloud (✓).

I'll put this to my backlog, I have other items that would benefit from external UX review. I don't consider this the final solution, only an improvement to make it easier to translate later.